airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/utils/asserts.py

```diff
@@ -67,7 +90,7 @@ def verify_records_schema(
     for stream in catalog.streams:
         schema_to_validate_against = stream.stream.json_schema
         if fail_on_extra_columns:
-            schema_to_validate_against = _enforce_no_additional_top_level_properties(schema_to_validate_against)
+            schema_to_validate_against = _enforce_false_additional_properties(schema_to_validate_against)
```

Comment:
I would find it more elegant to implement a custom [Validator](https://python-jsonschema.readthedocs.io/en/latest/api/jsonschema/protocols/#jsonschema.protocols.Validator) class which would enforce additional properties to false.

Reply:
agreed, moved schema transformation inside validator class